sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -245,6 +246,10 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
         "read files of Hive data source directly.")
     }
 
```

Comment:
What I propose is
```
def load(paths: String*): DataFrame = {
  if (extraOptions.contains("path") && paths.nonEmpty) fail ...
  ...
  if (DSV2) {
    val pathOption = if (paths.isEmpty) {
      None
    } else if (paths.length == 1) {
      Some("path" -> paths.head)
    } else {
      Some("paths" -> ...)
    }
  } else {
    // for DS V1, do not generate extra path option
  }
}
```

Reply:
Removing `def load(path: String)` breaks the binary compatibility:
```Exception in thread "main" java.lang.NoSuchMethodError: org.apache.spark.sql.DataFrameReader.load(Ljava/lang/String;)Lorg/apache/spark/sql/Dataset;```
So, I will not remove it.

Also, is the following required?
> else if (paths.length == 1) {
>       Some("path" -> paths.head)
>     }

I see `paths` should handle one or more paths: https://github.com/apache/spark/blob/8659ec554fe1993fcaf2ec4d1c4745205bddec7d/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileDataSourceV2.scala#L51-L57

Reply:
It's required, because
```
def load(path: String): DataFrame = {
  // force invocation of `load(...varargs...)`
  option("path", path).load(Seq.empty: _*)
}
```

If we remove this method and always call `def load(paths: String*)`, we should make sure the behavior doesn't change, and we still generate "path" option if only one path is given.

Reply:
Hmm, `def parquet(path: String)` has been calling `def load(paths: String*)`, not `def load(path: String)`, and it seems to have worked fine so far (other than the inconsistency this PR addresses), no?

Btw, we shouldn't remove `def load(path: String)` because it breaks the binary compatibility.

Reply:
parquet is a builtin source, so we can make it work anyway. `def load(path: String)` is general for other external sources, and we should guarantee backward compatibility here.

Reply:
Ah OK. Thanks.